File: 007: 10001st Prime/NthPrime.java
import java.util.Scanner;

public class NthPrime {

    /*
    By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see
    that the 6th prime is 13.

    What is the nth prime number?
     */
    public static void main(String[] args) {
        int count = 0;
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter which prime you want: ");
        int nth = scanner.nextInt();
        int nthPrime = 0;

        for (int i = 2; i < Integer.MAX_VALUE; i++) {
            boolean isPrime = true;
            for (int j = 2; j < i; j++) {
                if (i % j == 0) {
                    isPrime = false;
                    break;
                }
            }
            if (isPrime) {
                count++;
                nthPrime = i;
            }
            if (count == nth) {
                break;
            }
        }
        System.out.println(nthPrime);
    }
}

File: 007: 10001st Prime/NthPrime.py
import math
import sys


def main():
    count = 0
    nth = int(input("Enter which prime you want: "))
    nthPrime = 0
    max_int = sys.maxsize

    for i in range(2, max_int):
        isPrime = is_prime(i)

        if isPrime:
            count += 1
            nthPrime = i
            if nth == count:
                break

    print(nthPrime, "is the", nth, "prime number")


def is_prime(factor):
    prime = True
    for i in range(2, factor):
        if factor % i == 0:
            prime = False
            break
        i += 1

    return prime


main()
